Apps/WebClient/src/ClientApp/src/router.ts

```diff
@@ -360,6 +360,11 @@ export const beforeEachGuard: NavigationGuard = (
                 availableModules.includes(val)
             );
 
+        if (!hasRequiredModules) {
+            next({ path: UNAUTHORIZED_PATH });
+            return;
+        }
```

Comment:
After taking another look at the code, I think it would be cleaner to move this logic into the switch block for the case `UserState.unauthenticated`.  Something like:

```
if (hasRequiredModules) {
    next({ path: "/login", query: { redirect: to.path } });
} else {
    next({ path: UNAUTHORIZED_PATH });
}
```